main.tf

```diff
@@ -103,102 +105,71 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
         }
       }
 
       viewer_protocol_policy = default_cache_behavior.value.viewer_protocol_policy
-      min_ttl                = default_cache_behavior.value.min_ttl
-      default_ttl            = default_cache_behavior.value.default_ttl
-      max_ttl                = default_cache_behavior.value.max_ttl
+      min_ttl                = lookup(default_cache_behavior.value, "min_ttl", null)
+      default_ttl            = lookup(default_cache_behavior.value, "default_ttl", null)
+      max_ttl                = lookup(default_cache_behavior.value, "max_ttl", null)
     }
   }
 
   dynamic "ordered_cache_behavior" {
-    for_each = [for i in var.dynamic_ordered_cache_behavior : {
-      path_pattern                = i.path_pattern
-      allowed_methods             = i.allowed_methods
-      cached_methods              = i.cached_methods
-      target_origin_id            = i.target_origin_id
-      compress                    = i.compress
-      query_string                = i.query_string
-      cookies_forward             = i.cookies_forward
-      headers                     = i.headers
-      viewer_protocol_policy      = i.viewer_protocol_policy
-      min_ttl                     = i.min_ttl
-      default_ttl                 = i.default_ttl
-      max_ttl                     = i.max_ttl
-      lambda_function_association = lookup(i, "lambda_function_association", null)
-    }]
+    for_each = var.dynamic_ordered_cache_behavior
+    iterator = cache_behavior
     content {
-      path_pattern     = ordered_cache_behavior.value.path_pattern
-      allowed_methods  = ordered_cache_behavior.value.allowed_methods
-      cached_methods   = ordered_cache_behavior.value.cached_methods
-      target_origin_id = ordered_cache_behavior.value.target_origin_id
-      compress         = ordered_cache_behavior.value.compress
+      path_pattern     = cache_behavior.value.path_pattern
+      allowed_methods  = cache_behavior.value.allowed_methods
+      cached_methods   = cache_behavior.value.cached_methods
+      target_origin_id = cache_behavior.value.target_origin_id
+      compress         = lookup(cache_behavior.value, "compress", null)
 
       forwarded_values {
-        query_string = ordered_cache_behavior.value.query_string
+        query_string = cache_behavior.value.query_string
         cookies {
-          forward = ordered_cache_behavior.value.cookies_forward
+          forward = cache_behavior.value.cookies_forward
         }
-        headers = ordered_cache_behavior.value.headers
+        headers = lookup(cache_behavior.value, "headers", null)
       }
 
       dynamic "lambda_function_association" {
-        for_each = ordered_cache_behavior.value.lambda_function_association == null ? [] : [for i in ordered_cache_behavior.value.lambda_function_association : {
-          event_type   = i.event_type
-          lambda_arn   = i.lambda_arn
-          include_body = i.include_body
-        }]
```

Comment:
Also not sure this can be removed as you have potential to iterate over multiple associations (I think max 3)

Reply:
yup max 4 associations, locally tested with 2,1 and 0 it iterates